datasets/neural_code_search/neural_code_search.py

```diff
@@ -0,0 +1,168 @@
+                    "question_author_url": datasets.Value("string"),
+                    "answer": datasets.Value("string"),
+                    "answer_url": datasets.Value("string"),
+                    "answer_author": datasets.Value("string"),
+                    "answer_author_url": datasets.Value("string"),
+                    "examples": datasets.features.Sequence(datasets.Value("int32")),
+                    "examples_url": datasets.features.Sequence(datasets.Value("string")),
+                }
+            )
+        else:
+            features = datasets.Features(
+                {
+                    "id": datasets.Value("int32"),
+                    "filepath": datasets.Value("string"),
+                    "method_name": datasets.Value("string"),
+                    "start_line": datasets.Value("int32"),
+                    "end_line": datasets.Value("int32"),
+                    "url": datasets.Value("string"),
+                }
+            )
+
+        return datasets.DatasetInfo(
+            description=_DESCRIPTION,
+            features=features,
+            supervised_keys=None,
+            homepage=_HOMEPAGE,
+            license=_LICENSE,
+            citation=_CITATION,
+        )
+
+    def _split_generators(self, dl_manager):
+        """Returns SplitGenerators."""
+        my_urls = [url for config, url in _URLs.items() if config.startswith(self.config.name)]
+        data_dir = dl_manager.download_and_extract(my_urls)
+
+        return [
+            datasets.SplitGenerator(
+                name=datasets.Split.TRAIN,
+                gen_kwargs={
+                    "datapath": data_dir,
+                    "split": "train",
+                },
+            ),
+        ]
+
+    def _generate_examples(self, datapath, split):
+        """ Yields examples. """
+        for dp in datapath:
+            if self.config.name == "evaluation_dataset":
+                with open(dp, encoding="utf-8") as f:
+                    data = json.load(f)
+                    for id_, row in enumerate(data):
+                        yield id_, {
+                            "stackoverflow_id": row["stackoverflow_id"],
+                            "question": row["question"],
+                            "question_url": row["question_url"],
+                            "question_author": row["question_author"],
+                            "question_author_url": row["question_author_url"],
+                            "answer": row["answer"],
+                            "answer_url": row["answer_url"],
+                            "answer_author": row["answer_author"],
+                            "answer_author_url": row["answer_author_url"],
+                            "examples": row["examples"],
+                            "examples_url": row["examples_url"],
+                        }
+            else:
+                for dirpath, _, fnames in os.walk(dp):
```

Comment:
```suggestion
                for dirpath, _, fnames in sorted(os.walk(dp)):
```
we need to make the order deterministic here